src/BenchmarkDotNet/Environments/Jit.cs

```diff
@@ -2,6 +2,12 @@
 {
     public enum Jit
     {
+        /// <summary>
+        /// Undefined
+        /// <remarks>By default</remarks>
+        /// </summary>
+        Undefined,
```

Comment:
If user runs some benchmarks with following config:

```cs
DefaultConfig.Instance
	.With(Job.Default)
	.With(Job.RyuJitX64)
```

What is going to be printed in the `Jit` column for the default job? Is it going to be `Undefined` or `Default`?

@AndreyAkinshin  Maybe we should call this value a `Default` to avoid possible confusion? (users will ask what is Undefined Jit?)

Reply:
@adamsitnik 
If you do not set Jit in Benchmark configuration then benchmark will be ran with LegacyJIT (because it is the first element in Jit enum). 
In my branch I added the new enum element to avoid such behavior. If I run benchmark with your config I get such summary:

// * Summary *

BenchmarkDotNet=v0.10.14.20180523-develop, OS=Windows 10.0.17134
Intel Core i7-4710MQ CPU 2.50GHz (Haswell), 1 CPU, 8 logical and 4 physical cores
Frequency=2435775 Hz, Resolution=410.5470 ns, Timer=TSC
  [Host]     : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 64bit RyuJIT-v4.7.3101.0
  DefaultJob : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 64bit RyuJIT-v4.7.3101.0
  RyuJitX64  : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 64bit RyuJIT-v4.7.3101.0

Jit=RyuJit  Platform=X64  

 Method |       Job |      Mean |     Error |    StdDev |
------- |---------- |----------:|----------:|----------:|
   Test |   Default | 0.0020 ns | 0.0039 ns | 0.0035 ns |
   Test | RyuJitX64 | 0.0000 ns | 0.0000 ns | 0.0000 ns |

Reply:
@morgan-kn thanks for taking care of this! I remember that we used  to display `LegacyJit` in some places because of the default enum value ;)

Reply:
@adamsitnik I'm going to do post review changes. Should I rename Jit.Undefined to Jit.Default? I think it's a good idea. =)

Reply:
@morgan-kn I think the same, please change it. 👍